### ISSUE TITLE
Introduce functionality for sparse matrices in diffusion and degree_weight

### ISSUE THREAD

**Issue by zietzm** [greenelab/hetmech#43]
Subissue of #13 
Add support for scipy.sparse matrices in diffusion.py and degree_weight.py. Matrix type will for now be selected as a function parameter.
`matrix.normalize` can now output a sparse matrix, meaning that the dense intermediates should only be of dimension 1. Support allowing all matrices to be sparse

**Comment by dhimmel** [comment]
Something to keep in mind would be an `auto` mode where the matrices would be converted to dense arrays only once they exceed a density threshold.

**Comment by zietzm** [comment]
Had planned to implement something like that with a later issue and a new PR.

**Comment by zietzm** [comment]
Update:
A method called ```sparse_threshold``` has been implemented in #44 within [degree_weight.dwpc_duplicated_metanode](https://github.com/greenelab/hetmech/pull/44/files#diff-dcb7103539fb7297ec924e8232f7736eR120) and [matrix.metaedge_to_adjacency_matrix](https://github.com/greenelab/hetmech/pull/44/files#diff-2e95c8eb7e7141bd64ce5096a28d34b6R21). A user chooses the threshold above which the matrices should be dense, and conversion occurs whenever the matrix crosses the threshold. 

This method is implemented in two files so that (1.) new matrices will be created that are sparse or dense according to the threshold and (2.) as matrices are multiplied in a row and the density changes, the matrix will also be changed between sparse and dense according to the threshold.
